test/e2e/rte.go

```diff
@@ -14,71 +14,235 @@ See the License for the specific language governing permissions and
 
 	ginkgo.Context("with cluster configured", func() {
-		ginkgo.It("should fill the node resource topologies CR with the data", func() {
+		ginkgo.It("it should not account for any cpus if a container doesn't request exclusive cpus", func() {
+			ginkgo.By("getting the initial topology information")
+			initialNodeTopo := getNodeTopology(topologyClient, topologyUpdaterNode.Name, namespace)
+			ginkgo.By("creating a pood consuming the shared pool")
+			sleeperPod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sleeper-be-pod",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						v1.Container{
+							Name:  "sleeper-be-cnt",
+							Image: utils.CentosImage,
+							// 1 hour (or >= 1h in general) is "forever" for our purposes
+							Command: []string{"/bin/sleep", "1h"},
+						},
+					},
+				},
+			}
+
+			podMap := make(map[string]*v1.Pod)
+			pod := f.PodClient().CreateSync(sleeperPod)
+			podMap[pod.Name] = pod
+			defer utils.DeletePodsAsync(f, podMap)
+
+			// the object, hance the resource version must NOT change, so we can only sleep
+			waitSteps := 4
+			waitStepAmount := 5 * time.Second
+			for waitStep := 0; waitStep < waitSteps; waitStep++ {
+				ginkgo.By(fmt.Sprintf("getting the updated topology - wait for %v (%d/%d)", waitStepAmount, waitStep+1, waitSteps))
+				time.Sleep(waitStepAmount)
+			}
+			ginkgo.By("checking the changes in the updated topology - expecting none")
+			finalNodeTopo := getNodeTopology(topologyClient, topologyUpdaterNode.Name, namespace)
+
+			initialAllocRes := allocatableResourceListFromNodeResourceTopology(initialNodeTopo)
+			finalAllocRes := allocatableResourceListFromNodeResourceTopology(finalNodeTopo)
+			if len(initialAllocRes) == 0 || len(finalAllocRes) == 0 {
+				ginkgo.Fail(fmt.Sprintf("failed to find allocatable resources from node topology initial=%v final=%v", initialAllocRes, finalAllocRes))
+			}
+			zoneName, resName, cmp, ok := cmpAllocatableResources(initialAllocRes, finalAllocRes)
+			framework.Logf("zone=%q resource=%q cmp=%v ok=%v", zoneName, resName, cmp, ok)
+			if !ok {
+				ginkgo.Fail(fmt.Sprintf("failed to compare allocatable resources from node topology initial=%v final=%v", initialAllocRes, finalAllocRes))
+			}
+			gomega.Expect(cmp).To(gomega.Equal(0), fmt.Sprintf("final allocatable resources not equal - initial=%v final=%v", initialAllocRes, finalAllocRes))
+		})
+
+		ginkgo.It("it should account for containers requesting exclusive cpus", func() {
+			nodes, err := utils.FilterNodesWithEnoughCores(workerNodes, "1000m")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			if len(nodes) < 1 {
+				ginkgo.Skip("not enough allocatable cores for this test")
+			}
+
+			ginkgo.By("getting the initial topology information")
+			initialNodeTopo := getNodeTopology(topologyClient, topologyUpdaterNode.Name, namespace)
+			ginkgo.By("creating a pood consuming the shared pool")
+			sleeperPod := &v1.Pod{
```

Comment:
In general I prefer to optimize tests for readability rather than reusability, and in this case it still seems inlining the very simple pod definition is the clearer option